dvc/stage/cache.py

```diff
@@ -167,7 +172,7 @@ def save(self, stage):
         self.tree.move(PathInfo(tmp), path)
 
     def restore(self, stage, run_cache=True, pull=False):
-        if stage.is_callback or stage.always_changed:
+        if not _can_hash(stage):
```

Comment:
Better approach here would be to try to checkout/pull anything that is up-to-date(e.g. `dvc add`ed files and `always_changed` stages, but that needs to be done on a higher level. Will be resolved in https://github.com/iterative/dvc/issues/4742